api/sessionmngr/service_test.go

```diff
@@ -18,18 +19,65 @@ func TestListSessions(t *testing.T) {
 	ctx := context.TODO()
 
 	sessions := []models.Session{
-		{UID: "uid"},
+		{UID: "uid1"},
+		{UID: "uid2"},
+		{UID: "uid3"},
 	}
 
 	query := paginator.Query{Page: 1, PerPage: 10}
 
-	mock.On("SessionList", ctx, query).
-		Return(sessions, len(sessions), nil).Once()
+	Err := errors.New("error")
 
-	returnedSessions, count, err := s.ListSessions(ctx, query)
-	assert.NoError(t, err)
-	assert.Equal(t, sessions, returnedSessions)
-	assert.Equal(t, count, len(sessions))
+	type Expected struct {
+		sessions []models.Session
+		count    int
+		err      error
+	}
+
+	tests := []struct {
+		name          string
+		pagination    paginator.Query
+		ctx           context.Context
+		requiredMocks func()
+		expected      Expected
+	}{
+		{
+			name:       "ListSessions fails",
+			pagination: query,
+			ctx:        ctx,
+			requiredMocks: func() {
+				mock.On("SessionList", ctx, query).
+					Return(nil, 0, Err).Once()
+			},
+			expected: Expected{
+				sessions: nil,
+				count:    0,
+				err:      Err,
+			},
+		},
+		{
+			name:       "ListSessions succeeds",
+			pagination: query,
+			ctx:        ctx,
+			requiredMocks: func() {
+				mock.On("SessionList", ctx, query).
+					Return(sessions, len(sessions), nil).Once()
+			},
+			expected: Expected{
+				sessions: sessions,
+				count:    len(sessions),
+				err:      nil,
+			},
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
It seems the usual code style for this is

```go
for _, tc := range cases {
```

This will allow you to add the following to the linter config:

```yaml
issues:
  exclude:
     - Using the variable on range scope `tc` in function literal
```

So we don't need to whitelist those one by one.